arch/arm/core/mpu/cortex_m/mpu.c

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2025 STMicroelectronics
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/arch/arm/cortex_m/arm_core_mpu.h>
+
+#define LOG_LEVEL CONFIG_MPU_LOG_LEVEL
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(mpu_cortex_m);
+
+#if defined(CONFIG_CPU_CORTEX_M33) || defined(CONFIG_CPU_CORTEX_M23) || \
+	defined(CONFIG_CPU_CORTEX_M55) || defined(CONFIG_CPU_CORTEX_M85)
+	/* Armv8-M MPU: save RBAR and RLAR */
+#define ATTRIBUTE_AND_SIZE_REG_NAME RLAR
+#elif defined(CONFIG_CPU_CORTEX_M0PLUS) || defined(CONFIG_CPU_CORTEX_M3) || \
+	defined(CONFIG_CPU_CORTEX_M4) || defined(CONFIG_CPU_CORTEX_M7)
+	/* Classic MPU: save RBAR and RASR */
+#define ATTRIBUTE_AND_SIZE_REG_NAME RASR
+#else
+#error "Unsupported CortexM version"
+#endif
```

Comment:
With proposed changed in other file, this simplifies to:
```suggestion
#if Z_ARM_CPU_HAS_PMSAV8_MPU
#define ATTRIBUTE_AND_SIZE_REG_NAME RLAR
#else
#define ATTRIBUTE_AND_SIZE_REG_NAME RASR
#endif
```

----- 

It just occurs to me we could wrap this into a function instead. It might be considered cleaner, but is more verbose:
```c
static inline uint32_t read_mpu_rasr_rlar(void) {
#if Z_ARM_CPU_HAS_PMSAV8_MPU
    return MPU->RLAR;
#else
    return MPU->RASR;
#endif
}
/* ditto for write... */


/* in loops below */
   ctx->rasr_rlar[i] = read_mpu_rasr_rlar();
   /* OR */
   write_mpu_rasr_rlar(ctx->rasr_rlar[i]);
```